### PR TITLE
Add comprehensive image normalization documentation for wandb.Image DOCS-1016

### PR DESCRIPTION
**Pull request by mdlinville**
## Summary

This PR addresses [DOCS-1016](https://wandb.atlassian.net/browse/DOCS-1016) and the related GitHub issue [wandb/wandb#7377](https://github.com/wandb/wandb/issues/7377) by adding comprehensive documentation for `wandb.Image` normalization behavior.

This is part of a three-part project. The other two PRs are:

- New example Colab notebook: https://github.com/wandb/examples/pull/609
- `wandb/wandb` change: https://github.com/wandb/wandb/pull/10380

Ready for review.

### Documentation Updates (`docs` repository)

1. **Enhanced `Image.md` reference documentation** (`content/en/ref/python/sdk/data-types/Image.md`):
   - Added detailed "Image Normalization" section with sub-sections:
     - When Normalization is Applied (PyTorch tensors, NumPy arrays vs PIL Images, file paths)
     - Normalization Algorithm (explaining `_guess_and_rescale_to_0_255` logic)
     - Examples of Normalization Effects (with code snippets for different input ranges)
     - Best Practices
     - Common Pitfalls

2. Added Image normalization guide (new page) with some more details.
3. **Updated media logging guide** (`content/en/guides/models/track/log/media.md`):
   - Modified existing text to reference the new detailed normalization section

4. **Translated documentation** (Japanese and Korean):
   - Updated `content/ja/ref/python/data-types/image.md` and `content/ko/ref/python/data-types/image.md` with translated normalization sections
   - Updated media guides in both languages to reference the new sections
   - Fixed broken `relref` links that were pointing to incorrect paths

### SDK Documentation Updates (`wandb/wandb` repository)

See https://github.com/wandb/wandb/pull/10380

## Problem Solved

The original issue was that users found the existing documentation about image normalization to be "cryptic" and unclear. Users were wasting computation time due to misunderstandings about:
- When normalization occurs
- How the normalization algorithm works
- What input ranges are supported
- How to avoid unwanted normalization

## Testing

- Hugo build completes successfully with no errors
- All `relref` links are properly resolved
- Documentation is available in English, Japanese, and Korean
- Developed the notebook in  https://github.com/wandb/examples/pull/609 and ran the cells

## Related PRs

- Companion PR in `wandb/wandb` repository: [wandb/wandb#10380](https://github.com/wandb/wandb/pull/10380)
- Companior PR in `wandb/examples` repository: https://github.com/wandb/examples/pull/609

## JIRA Reference

- [DOCS-1016](https://wandb.atlassian.net/browse/DOCS-1016): "Productboard: [Docs] Better explanation for the section: when logging a PyTorch tensor as a PIL Image"

📄 **[View preview links for changed pages](https://github.com/wandb/docs/pull/1559#issuecomment-3215338217)**

[DOCS-1016]: https://wandb.atlassian.net/browse/DOCS-1016?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DOCS-1016]: https://wandb.atlassian.net/browse/DOCS-1016?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by mdlinville** [comment]
**Update**: Companion PR in `wandb/wandb` repository: [wandb/wandb#10380](https://github.com/wandb/wandb/pull/10380)

File: content/en/guides/models/track/log/media.md [hunk 1]
```diff
@@ -58,7 +58,7 @@ with wandb.init(project="image-log-example") as run:
...
+We assume the image is gray scale if the last dimension is 1, RGB if it's 3, and RGBA if it's 4. If the array contains floats, we automatically normalize them to integers between `0` and `255` using our normalization algorithm. For detailed information about how normalization works with PyTorch tensors and NumPy arrays, see the [Image Normalization section]({{< relref "/ref/python/sdk/data-types/image.md#image-normalization" >}}) in the Image reference. If you want to normalize your images differently, you can specify the [`mode`](https://pillow.readthedocs.io/en/stable/handbook/concepts.html#modes) manually or just supply a [`PIL.Image`](https://pillow.readthedocs.io/en/stable/reference/Image.html), as described in the "Logging PIL Images" tab of this panel.
```
Comment (mdlinville) [inline content/en/guides/models/track/log/media.md]: Avoid "we". Reword this to be more imperative.

File: content/en/ref/python/sdk/data-types/Image.md [hunk 2]
```diff
@@ -103,6 +103,103 @@ with wandb.init() as run:
     run.log({"examples": examples})
 ``` 
 
+## Image Normalization
```
Comment (mdlinville) [inline content/en/ref/python/sdk/data-types/Image.md]: Use sentence case for headings

File: content/en/ref/python/sdk/data-types/Image.md [hunk 3]
```diff
@@ -103,6 +103,103 @@ with wandb.init() as run:
...
+
+When you pass PyTorch tensors or NumPy arrays to `wandb.Image`, the pixel values are automatically normalized to the range [0, 255] unless you set `normalize=False`. This normalization is designed to handle common image data formats and ensure proper display.
```
Comment (mdlinville) [inline content/en/ref/python/sdk/data-types/Image.md]:
```suggestion
When you pass PyTorch tensors or NumPy arrays to `wandb.Image`, the pixel values are automatically normalized to the range [0, 255] unless you set `normalize=False`. This normalization is designed to handle and ensure proper display of common image formats.
```

File: content/en/ref/python/sdk/data-types/Image.md [hunk 4]
```diff
@@ -103,6 +103,103 @@ with wandb.init() as run:
...
+import wandb
+
+# Create tensor with values in [0, 1] range
+tensor_0_1 = torch.rand(3, 64, 64)
+
+# Disable normalization - values will be clipped to [0, 255]
+image = wandb.Image(tensor_0_1, normalize=False, caption="Normalization disabled")
+```
+
+### Best Practices
```
Comment (mdlinville) [inline content/en/ref/python/sdk/data-types/Image.md]: Recommendations

File: content/en/ref/python/sdk/data-types/Image.md [hunk 5]
```diff
@@ -103,6 +103,103 @@ with wandb.init() as run:
...
+
+1. **For consistent results**: Pre-process your data to the expected [0, 255] range before logging
+2. **To avoid normalization**: Convert tensors to PIL Images using `PILImage.fromarray()`
+3. **For debugging**: Use `normalize=False` to see the raw values (they will be clipped to [0, 255])
+4. **For precise control**: Use PIL Images when you need exact pixel values
+
+### Common Pitfalls
```
Comment (mdlinville) [inline content/en/ref/python/sdk/data-types/Image.md]: Troubleshooting

File: content/en/guides/models/track/log/media.md [hunk 6]
```diff
@@ -58,7 +58,7 @@ with wandb.init(project="image-log-example") as run:
...
+The system assumes the image is gray scale if the last dimension is 1, RGB if it's 3, and RGBA if it's 4. If the array contains floats, the system automatically normalizes them to integers between `0` and `255` using the normalization algorithm. For detailed information about how normalization works with PyTorch tensors and NumPy arrays, see the [Image Normalization section]({{< relref "/ref/python/sdk/data-types/image.md#image-normalization" >}}) in the Image reference. If you want to normalize your images differently, you can specify the [`mode`](https://pillow.readthedocs.io/en/stable/handbook/concepts.html#modes) manually or just supply a [`PIL.Image`](https://pillow.readthedocs.io/en/stable/reference/Image.html), as described in the "Logging PIL Images" tab of this panel.
```
Comment (mdlinville) [inline content/en/guides/models/track/log/media.md]:
```suggestion
The system assumes the image is gray scale if the last dimension is 1, RGB if it's 3, and RGBA if it's 4. If the array contains floats, the system automatically normalizes them to integers between `0` and `255`. For detailed information about normalization with PyTorch tensors and NumPy arrays, see the [Image normalization section]({{< relref "/ref/python/sdk/data-types/image.md#image-normalization" >}}) in the Image SDK reference. To normalize your images differently, you can specify the [`mode`](https://pillow.readthedocs.io/en/stable/handbook/concepts.html#modes) manually or supply a [`PIL.Image`](https://pillow.readthedocs.io/en/stable/reference/Image.html), as described in the "Logging PIL Images" tab.
```

File: content/en/ref/python/sdk/data-types/Image.md [hunk 7]
```diff
@@ -103,6 +103,103 @@ with wandb.init() as run:
     run.log({"examples": examples})
 ``` 
 
+## Image normalization
+
+When you pass PyTorch tensors or NumPy arrays to `wandb.Image`, the pixel values are automatically normalized to the range [0, 255] unless you set `normalize=False`. This normalization is designed to handle and ensure proper display of common image formats.
+
+### When Normalization is Applied
```
Comment (mdlinville) [inline content/en/ref/python/sdk/data-types/Image.md]: Use sentence case for all headings

**Review by ngrayluna — COMMENTED** [review]
Thanks! Couple of questions, comments:
* Did you test out the code snippets generated?
* You'll want to make a PR for changes added in the ref page

File: content/en/ref/python/sdk/data-types/Image.md [hunk 8]
```diff
@@ -103,6 +103,103 @@ with wandb.init() as run:
     run.log({"examples": examples})
 ``` 
 
+## Image normalization
```
Comment (ngrayluna) [inline content/en/ref/python/sdk/data-types/Image.md]: Remove "#" signs for docs intended for docstrings. They don't render correctly.

File: content/en/ref/python/sdk/data-types/Image.md [hunk 9]
```diff
@@ -103,6 +103,103 @@ with wandb.init() as run:
...
+import wandb
+
+# Create tensor with values in [0, 1] range
+tensor_0_1 = torch.rand(3, 64, 64)
+
+# Disable normalization - values will be clipped to [0, 255]
+image = wandb.Image(tensor_0_1, normalize=False, caption="Normalization disabled")
+```
+
+### Recommendations
```
Comment (ngrayluna) [inline content/en/ref/python/sdk/data-types/Image.md]: These sections look like they'd be better suited for the dev guide, not the reference guide.

**Review by ngrayluna — COMMENTED** [review]
This content is better suited for the dev guide.  You want to keep the docstrings as concise as possible. They should include code examples, but that's it; (i.e. no markdown sections , etc.)

**Comment by mdlinville** [comment]
Test comment to verify GitHub MCP server is working properly.

File: content/en/guides/models/track/log/image-normalization.md [hunk 10]
```diff
@@ -0,0 +1,139 @@
+---
+title: "Image Normalization Guide"
+description: "Learn how wandb.Image handles normalization for different input types and how to control this behavior"
+---
+
+# Image Normalization Guide
```
Comment (mdlinville) [inline content/en/guides/models/track/log/image-normalization.md]:
```suggestion
```

File: content/en/guides/models/track/log/image-normalization.md [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+
+When you pass PyTorch tensors or NumPy arrays to `wandb.Image`, the pixel values are automatically normalized to the range [0, 255] unless you set `normalize=False`. This guide explains how normalization works and how to control it.
```
Comment (mdlinville) [inline content/en/guides/models/track/log/image-normalization.md]:
```suggestion
When you pass PyTorch tensors or NumPy arrays to `wandb.Image`, the pixel values are automatically normalized to the range [0, 255] unless you set `normalize=False`. This guide explains how image normalization works and how to control it.
```

File: content/en/guides/models/track/log/image-normalization.md [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+
+## When normalization is applied
+
+Normalization is applied to:
+- **PyTorch tensors** (format: `(channel, height, width)`)
+- **NumPy arrays** (format: `(height, width, channel)`)
+
+Normalization is **NOT** applied to:
+- **PIL Images** (passed as-is)
+- **File paths** (loaded as-is)
```
Comment (mdlinville) [inline content/en/guides/models/track/log/image-normalization.md]: Consider converting this to a table using ✅ and ❌ .

File: content/en/guides/models/track/log/image-normalization.md [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+
+2. **If data is in range [-1, 1]**: Values are rescaled to [0, 255] using:
+   ```python
+   normalized_data = (255 * 0.5 * (data + 1)).astype(np.uint8)
+   ```
+
+3. **For any other range**: Values are clipped to [0, 255] and converted to uint8
+   ```python
+   normalized_data = data.clip(0, 255).astype(np.uint8)
+   ```
```
Comment (mdlinville) [inline content/en/guides/models/track/log/image-normalization.md]: Change these to bullets

File: content/en/guides/models/track/log/image-normalization.md [hunk 14]
```diff
@@ -0,0 +1,139 @@
...
+```
+
+**Note on visual contrast**: The [-1, 1] normalization creates higher visual contrast compared to [0, 1] normalization. This is because:
+- Negative values (like -0.8) become very dark (around 25)
+- Positive values (like 0.8) become very bright (around 230)
+- Values near 0 become mid-gray (127.5)
+
+This "stretches" the visual range, making differences between pixel values more pronounced. This is particularly useful for highlighting subtle patterns in machine learning data, but if you want less contrast, consider preprocessing your data to a [0, 1] range before logging.
+
+### Example 3: Avoiding normalization with PIL Images
```
Comment (mdlinville) [inline content/en/guides/models/track/log/image-normalization.md]:
```suggestion
### Example 3: Avoid normalization with PIL Images
```

File: content/en/guides/models/track/log/image-normalization.md [hunk 15]
```diff
@@ -0,0 +1,139 @@
...
+
```
Comment (mdlinville) [inline content/en/guides/models/track/log/image-normalization.md:74]:
```suggestion
Normalization is not applied to PIL Images.

```

File: content/en/guides/models/track/log/image-normalization.md [hunk 16]
```diff
@@ -0,0 +1,139 @@
...
+# Create tensor with values in [0, 1] range
+tensor_0_1 = torch.rand(3, 64, 64)
+
+# Convert to PIL Image to avoid normalization
+pil_image = PILImage.fromarray((tensor_0_1.permute(1, 2, 0).numpy() * 255).astype('uint8'))
+image = wandb.Image(pil_image, caption="No normalization applied")
+```
+
+### Example 4: Using normalize=False
+
```
Comment (mdlinville) [inline content/en/guides/models/track/log/image-normalization.md:90]:
```suggestion
To explicitly turn off image normalization without converting the image, set `normalize=False`.

```

File: content/en/guides/models/track/log/image-normalization.md [hunk 17]
```diff
@@ -0,0 +1,139 @@
...
+- Your data is already in the correct range (like [0, 255] integers)
+- You're debugging normalization issues
+- Quick testing without additional processing steps
+
+### Use automatic normalization when:
+- You want consistent behavior across different input types
+- Your data is in standard ranges ([0, 1] or [-1, 1])
+- You want the system to handle the conversion automatically
+
+## Best practices
```
Comment (mdlinville) [inline content/en/guides/models/track/log/image-normalization.md]: Combine the Best practices sectino with the Common issues section and call it Troubleshooting. You will need to adjust the contents a bit.

File: content/en/guides/models/track/log/media.md [hunk 18]
```diff
@@ -58,7 +58,7 @@ with wandb.init(project="image-log-example") as run:
...
 ```
 
-We assume the image is gray scale if the last dimension is 1, RGB if it's 3, and RGBA if it's 4. If the array contains floats, we convert them to integers between `0` and `255`. If you want to normalize your images differently, you can specify the [`mode`](https://pillow.readthedocs.io/en/stable/handbook/concepts.html#modes) manually or just supply a [`PIL.Image`](https://pillow.readthedocs.io/en/stable/reference/Image.html), as described in the "Logging PIL Images" tab of this panel.   
+The system assumes the image is gray scale if the last dimension is 1, RGB if it's 3, and RGBA if it's 4. If the array contains floats, the system automatically normalizes them to integers between `0` and `255`. For detailed information about normalization with PyTorch tensors and NumPy arrays, see the [Image Normalization Guide]({{< relref "/guides/models/track/log/image-normalization.md" >}}). To normalize your images differently, you can specify the [`mode`](https://pillow.readthedocs.io/en/stable/handbook/concepts.html#modes) manually or supply a [`PIL.Image`](https://pillow.readthedocs.io/en/stable/reference/Image.html), as described in the "Logging PIL Images" tab.
```
Comment (mdlinville) [inline content/en/guides/models/track/log/media.md]: Use "W&B" instead of "the system". Use sentence case with initial cap for the link text.

File: content/en/guides/models/track/log/image-normalization.md [hunk 19]
```diff
@@ -0,0 +1,141 @@
...
+---
+
+
+When you pass PyTorch tensors or NumPy arrays to `wandb.Image`, the pixel values are automatically normalized to the range [0, 255] unless you set `normalize=False`. This guide explains how image normalization works and how to control it.
+
+## When normalization is applied
+
+| Input Type | Format | Normalization Applied | Notes |
+|------------|--------|----------------------|-------|
+| **PyTorch tensors** | `(channel, height, width)` | ✅ Yes | Automatically normalized to [0, 255] range |
```
Comment (ngrayluna) [inline content/en/guides/models/track/log/image-normalization.md:13]: Remove emojis

File: content/ja/guides/integrations/pytorch.md [hunk 20]
```diff
@@ -49,7 +49,7 @@ for batch_idx, (data, target) in enumerate(train_loader):
 
 ## 画像とメディアのログ
 
-画像データを持つ PyTorch `Tensors` を [`wandb.Image`]({{< relref path="/ref/python/data-types/image.md" lang="ja" >}}) に渡すことができ、[`torchvision`](https://pytorch.org/vision/stable/index.html) のユーティリティが自動的に画像に変換します。
```
Comment (ngrayluna) [inline content/ja/guides/integrations/pytorch.md:52]: Suggest undoing changes to JA and KO docs. They will get overwritten. Also, keeps the PR cleaner for review.

File: content/en/ref/python/sdk/data-types/Image.md [hunk 21]
```diff
@@ -9,10 +9,10 @@ data_type_classification: class
 
 
 
-## <kbd>class</kbd> `Image`
+<kbd>class</kbd> `Image`
```
Comment (ngrayluna) [inline content/en/ref/python/sdk/data-types/Image.md:12]: Formatting is correct. Undo changes. (Note: Anything in content/en/ref/python/* autogenerated. To make changes, make a PR in wandb/wandb).

**Review by ngrayluna — CHANGES_REQUESTED** [review]

**Comment by mdlinville** [comment]
This PR is too low quality to keep, I'll make another attempt once we are on Mintlify.
